lms/templates/dashboard.html

```diff
@@ -160,21 +160,25 @@
                   if is_fulfilled_entitlement:
                     # If the user has a fulfilled entitlement, pass through the entitlements CourseEnrollment object
                     enrollment = entitlement_session
+                    # We only show email settings for entitlement cards if the entitlement has an associated enrollment
+                    show_email_settings = (entitlement_session.course_id in show_email_settings_for)
                   else:
                     # If the user has an unfulfilled entitlement, pass through a bare CourseEnrollment object built off of the next available session
                     upcoming_sessions = course_entitlement_available_sessions[str(entitlement.uuid)]
                     next_session = upcoming_sessions[0] if upcoming_sessions else None
                     if not next_session:
                         continue
                     enrollment = CourseEnrollment(user=user, course_id=next_session['key'], mode=next_session['type'])
+                  
 
                 session_id = enrollment.course_id
                 show_courseware_link = (session_id in show_courseware_links_for)
                 cert_status = cert_statuses.get(session_id)
                 can_refund_entitlement = entitlement and entitlement.is_entitlement_refundable()
                 can_unenroll = (not cert_status) or cert_status.get('can_unenroll') if not unfulfilled_entitlement else False
                 credit_status = credit_statuses.get(session_id)
-                show_email_settings = (session_id in show_email_settings_for)
+                if not entitlement:
```

Comment:
this can be:

show_email_settings = (session_id in show_email_settings_for) if not entitlement else False

Reply:
(also in the themed one)